drivers/flash/flash_stm32l5x.c

```diff
@@ -21,8 +21,7 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 #include "flash_stm32.h"
 
 #if defined(CONFIG_SOC_SERIES_STM32H5X)
-/* at this time stm32h5 mcus have 128KB (stm32h50x) or 2MB (stm32h56x/57x) */
-#define STM32_SERIES_MAX_FLASH	2048
+#define STM32_SERIES_MAX_FLASH	(CONFIG_FLASH_SIZE)
```

Comment:
You could group H5 with U5 below to avoid duplication

Reply:
yes expect that comment does not apply for stm32H5